scala/lambdas/ingest-mapper/src/main/scala/uk/gov/nationalarchives/ingestmapper/MetadataService.scala

```diff
@@ -88,7 +88,26 @@ class MetadataService(s3: DAS3Client[IO], discoveryService: DiscoveryService) {
           Stream.evals {
             topLevelIdsToDepartmentSeries.map { itemToDepartmentSeries =>
               val parentPaths = getParentPaths(json, itemToDepartmentSeries)
-              val updatedJson = jsonArr.map { metadataEntry =>
+              val childCountMap = parentPaths.groupBy(_._2).view.mapValues(_.size).toMap
+
+              def addChildCount(obj: Obj) = {
+                val values = obj.value.toMap
+                val id = values("id").str
+                val parentPath = values.get("parentPath").map(parentPath => s"${parentPath.str}/").getOrElse("")
+                val objId = s"$parentPath$id"
```

Comment:
Maybe `pathToObj` or `objPath`?

Reply:
This is the parent path of the children so I've called it `childParentPath`